app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/MainActivity.java

```diff
@@ -0,0 +1,220 @@
+                    }
+
+                    phoneCursor.close();
+
+                    // Query and loop for every email of the contact
+                    Cursor emailCursor = contentResolver.query(EmailCONTENT_URI, null, EmailCONTACT_ID + " = ?", new String[]{contact_id}, null);
+
+                    while (emailCursor.moveToNext()) {
+
+                        email = emailCursor.getString(emailCursor.getColumnIndex(DATA));
+
+                        output.append("\nEmail:" + email);
+
+                        contact.setEmail(email);
+
+                    }
+
+                    ContactDatabaseHelper dbHelper = ContactDatabaseHelper.getInstance(this);
+                    db = dbHelper.getWritableDatabase();
+                    cupboard().withDatabase(db).put(contact);
+
+                    emailCursor.close();
+                }
+                output.append("\n");
+            }
+
+
+            outputText.setText(output);
+        }
+    }
+
+    private String getContactId(String _ID, Cursor cursor) {
+        return cursor.getString(cursor.getColumnIndex(_ID));
+    }
+
+    private void splitFirstAndLastName(String name, Contact contact) {
+        if (name.trim().length() > 0) {
+            String lastName = "";
+            String firstName;
+            if (name.split("\\s+").length > 1) {
+                try{
+                    lastName = name.substring(name.lastIndexOf(" ") + 1);
+                    firstName = name.substring(0, name.lastIndexOf(' '));
+                } catch (Exception exception){
+                    throw exception;
+                }
+            } else {
+                firstName = name;
+            }
+            contact.setFirstName(firstName);
+            contact.setLastName(lastName);
+        }
+    }
+
+    private String simplifyPhoneNumber(String phoneNumber) {
+        return phoneNumber.replaceAll("[()\\s-]+", "");
+    }
+
+    synchronized private void checkPermissions() {
+        if (ContextCompat.checkSelfPermission(this,
+                Manifest.permission.READ_CONTACTS)
+                != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, permissions, 3);
+        }
+    }
+
+    // Display all the contacts with numbers
+    private List<ContactItems> displayContacts() {
+
+        List<ContactItems> contactItemsList = new ArrayList<>();
+        ContentResolver cr = getContentResolver();
+        Cursor cur = cr.query(ContactsContract.Contacts.CONTENT_URI, null, null, null, null);
+        if (cur.getCount() > 0) {
+            while (cur.moveToNext()) {
+                String id = cur.getString(cur.getColumnIndex(ContactsContract.Contacts._ID));
+                String name = cur.getString(cur.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME));
+                String phoneNo = "";
+                if (Integer.parseInt(cur.getString(cur.getColumnIndex(ContactsContract.Contacts.HAS_PHONE_NUMBER))) > 0) {
```

Comment:
If this is the criteria for adding the contact, then don't populate id or name until you're in here